src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionProviderFactory.cs

```diff
@@ -12,6 +14,8 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
 {
     internal static class AddUsingsCodeActionProviderFactory
     {
+        private static readonly Regex AddUsingVSCodeAction = new Regex("using (.+);", RegexOptions.Compiled, TimeSpan.FromSeconds(1));
```

Comment:
1. won't work for VB.  
2. in hte case of `using static Foo;` the `(.+)` portoin will grab `static Foo`.  Is that what you want?
3. consider just string manipulation which is likely to be much faster.

Reply:
also, probably a good idea to add $ and ^ anchors here.

Reply:
> won't work for VB.

I believe that's fine, VB isn't a use case for Razor.

> in hte case of using static Foo; the (.+) portoin will grab static Foo. Is that what you want?

I believe so, we should be allowing that through. Added a test case 😄